### ISSUE TITLE
change name characteristic name, crash

### ISSUE THREAD

**Issue by sdetweil** [arduino-libraries/ArduinoBLE#194]
I want to be able to set the name use by the library in the GATT setServiceName() function.  (used in GATT.begin)

I added another 'GATT.begin' that takes a string (const char *)  and from the original, call the new one with "Arduino" 
which was hard coded before.. 

now the library crashes  in GATT.addService() (called from begin)

i added debugging 
```
10:03:36.109 -> gatt.begin, no name supplied
10:03:36.109 -> in gatt begin with name=Arduino
10:03:36.109 -> in gatt begin done with setDeviceName =Arduino
10:03:36.109 -> done with gatt.setAppearance
10:03:36.109 -> done with gatt.clearAttributes
10:03:36.109 -> entered gatt.addService
10:03:36.109 -> gatt.addService after retain
10:03:36.109 -> gatt.addService after attributes.add
10:03:36.109 -> gatt.addService pre loop1
10:03:36.109 -> gatt.addService post loop1
10:03:36.109 -> done with gatt.addService
10:03:36.109 -> entered gatt.addService
10:03:36.109 -> gatt.addService after retain
```
code 
```
void GATTClass::begin()
{
     if(_debug){
    _debug->println("gatt.begin, no name supplied");
  }
     begin("Arduino");
}
void GATTClass::begin(const char* name)
{  
  if(_debug){
    _debug->print("in gatt begin with name=");
    _debug->println(name);
  }
  setDeviceName(name);
  // old code was  setDeviceName("Arduino");   // <------------- old before change, called with var on stack, still on stack
  if(_debug){
    _debug->print("in gatt begin done with setDeviceName=");
    _debug->println(name);
  }
  setAppearance(0x000);
   if(_debug){
    _debug->println("done with gatt.setAppearance");
  }
  clearAttributes();
   if(_debug){
    _debug->println("done with gatt.clearAttributes");
  }
  addService(&_genericAccessService);
     if(_debug){
    _debug->println("done with gatt.addService");
  }
  addService(&_genericAttributeService);    // this one crashes.. 
   if(_debug){
    _debug->println("done with gatt.begin");
  }
}
```

i didn't change any other code.. (crashed before adding debug, and before adding debug to addService()
```
void GATTClass::addService(BLELocalService* service)
{
  if(_debug){
    _debug->println("entered gatt.addService");
  }
  service->retain();
  if(_debug){
    _debug->println("gatt.addService after retain");
  }
  _attributes.add(service);
  if(_debug){
    _debug->println("gatt.addService after attributes.add");
  }
  uint16_t startHandle = attributeCount();
  if(_debug){
    _debug->println("gatt.addService pre loop1");
  }
  for (unsigned int i = 0; i < service->characteristicCount(); i++) {
    BLELocalCharacteristic* characteristic = service->characteristic(i);

    characteristic->retain();
    _attributes.add(characteristic);
    characteristic->setHandle(attributeCount());
    
    // add the characteristic again to make space of the characteristic value handle
    _attributes.add(characteristic);
    if(_debug){
      _debug->println("gatt.addService pre loop2");
    }
    for (unsigned int j = 0; j < characteristic->descriptorCount(); j++) {
      BLELocalDescriptor* descriptor = characteristic->descriptor(j);

      descriptor->retain();
      _attributes.add(descriptor);
      descriptor->setHandle(attributeCount());
    }
    if(_debug){
      _debug->println("gatt.addService post loop2");
    }
  }
  if(_debug){
    _debug->println("gatt.addService post loop1");
  }
  service->setHandles(startHandle, attributeCount());
}
```

this should have been simple, what am I missing?

**Comment by sdetweil** [comment]
the proposed fix for #192 causes the crash.   remove the 5 added lines in 192 and all is ok. 

i want the mac address, and so I have to BLE. begin() to get it.


but I want tto set  the name to the text of the mac address


whuch I have to do on bBLE.begin

**Comment by sdetweil** [comment]
never mind
